src/mscorlib/src/System/String.cs

```diff
@@ -973,17 +973,20 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
 
             if ((count == 0) || (omitEmptyEntries && this.Length == 0)) 
             {           
-                return new String[0];
+                return EmptyArray<String>.Value;
```

Comment:
how's about using a static readonly string[] which is conditionally compiled to either `new string[0];`, and `EmptyArray<string>.Value`?